app/Actions/Album/Archive.php

```diff
@@ -229,7 +230,7 @@ private static function isArchivable(AbstractAlbum $album): bool
 	{
 		return
 			$album->is_downloadable ||
-			($album instanceof BaseSmartAlbum && AccessControl::is_logged_in()) ||
-			($album instanceof BaseAlbum && AccessControl::is_current_user_or_admin($album->owner_id));
+			($album instanceof BaseSmartAlbum && Auth::check()) ||
+			($album instanceof BaseAlbum && Gate::check(PhotoPolicy::OWN, $album));
```

Comment:
At least it should be

```suggestion
			($album instanceof BaseSmartAlbum && Auth::check()) ||
			($album instanceof BaseAlbum && Gate::check(AlbumPolicy::OWN, $album));
```

However, I wonder why you haven't refactored the whole method into an `ARCHIVE` policy. I guess I know what happened I which would also explain the previously spotted change in behavior.

There is a newly added `AlbumPolicy::download` which is used nowhere this is already suspicious.
Moreover we have a newly added `PhotoPolicy::download` which has been copied over from former `PhotoAuthorizationProvider::isDownloadable` without changes. (This also explains where call to `isVisible` comes from as it has always been there.)

However, above you now use `PhotoPolicy::download` instead of something like `PhotoPolicy::archive` which has a different logic. (This explains the change in behavior.) So what we would need in order to restore the old behavior would be a `AlbumPolicy::archive` and `PhotoPolicy::archive` with the same logic as before.